apps/wallets/smart-wallet/react/src/app/utils/createAAWallet/createW3AAAWallet.ts

```diff
@@ -13,41 +14,30 @@ export const createW3AAAWallet = async (isProd: boolean) => {
         throw new Error("No JWT token found");
     }
 
-    const { email } = parseToken(jwt);
-
     const xm = isProd
         ? SmartWalletSDK.init({
               clientApiKey: process.env.REACT_APP_CROSSMINT_API_KEY_PROD || "",
           })
         : SmartWalletSDK.init({
               clientApiKey: process.env.REACT_APP_CROSSMINT_API_KEY_STG || "",
           });
-
-    const userIdentifier = { id: email, jwt };
-
-    const web3AuthSigner: Web3AuthSigner = {
-        type: "WEB3_AUTH",
-        clientId: (isProd
-            ? process.env.REACT_APP_WEB3_AUTH_CLIENT_ID_PROD
-            : process.env.REACT_APP_WEB3_AUTH_CLIENT_ID_STG) as string,
-        web3AuthNetwork: (isProd
-            ? process.env.REACT_APP_WEB3_AUTH_NETWORK_PROD
-            : process.env.REACT_APP_WEB3_AUTH_NETWORK_STG) as Web3AuthSigner["web3AuthNetwork"],
-        verifierId: (isProd
-            ? process.env.REACT_APP_WEB3_AUTH_VERIFIER_ID_PROD
-            : process.env.REACT_APP_WEB3_AUTH_VERIFIER_ID_STG) as string,
-        jwt,
+    const chain = isProd ? Blockchain.POLYGON : Blockchain.POLYGON_AMOY;
+
+    const walletConfig = {
+        signer: await getWeb3AuthSigner({
+            clientId: (isProd
+                ? process.env.REACT_APP_WEB3_AUTH_CLIENT_ID_PROD
+                : process.env.REACT_APP_WEB3_AUTH_CLIENT_ID_STG) as string,
+            web3AuthNetwork: (isProd
+                ? process.env.REACT_APP_WEB3_AUTH_NETWORK_PROD
+                : process.env.REACT_APP_WEB3_AUTH_NETWORK_STG) as TORUS_NETWORK_TYPE,
+            verifierId: (isProd
+                ? process.env.REACT_APP_WEB3_AUTH_VERIFIER_ID_PROD
+                : process.env.REACT_APP_WEB3_AUTH_VERIFIER_ID_STG) as string,
+            jwt,
+            chain,
+        }),
     };
```

Comment:
I'm so hyped to kill this weird pattern where we have both the prod and staging env present. Kinda defeats the purpose lol.